lib/WUI/nhttp/req_parser.cpp

```diff
@@ -241,8 +270,159 @@ bool RequestParser::uri_filename(char *buffer, size_t buffer_size) const {
     return true;
 }
 
-bool RequestParser::authenticated() const {
-    return (holds_alternative<bool>(auth_status) && get<bool>(auth_status));
+RequestParser::NonceStatus RequestParser::nonce_status(uint64_t nonce_to_check) const {
+    uint32_t random = static_cast<uint32_t>(nonce_to_check >> 32);
+    uint32_t time = nonce_to_check & 0xffffffff;
+    auto age = ticks_s() - time;
+    if (random == nonce_random && age >= 0) {
```

Comment:
I don't think this is the case, because for any authentication to happen we need to call  the `authenticated_status()` function and it always checks if the random is 0 and if so it generates the random before calling `nonce_status`. This might only be a problem if the code would move around and this function would be used somewhere else.